File: dayTwo/Python/book_suggestion_system.py
import random

def get_suggestion(books):

    random_book = random.randint(0, len(books) - 1)
    random_page = random.randint(1, 100)

    return "\nBook for the Day\n" + "Book Title: " + books[random_book] + "\nPage: " + str(random_page)


def add_book(books, new_book):

    found = False

    for count in range(len(books)):

        if books[count].lower() == new_book.lower():
            found = True

    if found:
        return "Book already exists!"

    books.append(new_book)
    return "Book added successfully!"


def remove_book(books, book_title):

    for count in range(len(books)):

        if books[count].lower() == book_title.lower():

            books.pop(count)
            return "Book removed successfully!"

    return "Book not found!"


def update_book(books, old_title, new_title):

    for count in range(len(books)):

        if books[count].lower() == old_title.lower():

            books[count] = new_title
            return "Book updated successfully!"

    return "Book not found!"


def show_books(books):

    all_books = "\nAll Books\n"

    for count in range(len(books)):

        all_books = all_books + str(count + 1) + ". " + books[count] + "\n"

    return all_books


books = []

books.append("The Hobbit")
books.append("The Mystery")
books.append("Animal Farm")
books.append("Brave Kingdom")

operation = 0

while operation != 6:

    print("\nWelcome to the Book Suggestion System!")
    print("1. Get Suggestions")
    print("2. Add Book")
    print("3. Remove Book")
    print("4. Update Book")
    print("5. Show All Books")
    print("6. Exit")


    user_input = input("Enter operation: ")


    match user_input:
        case "1":
            answer = "yes"
            while answer.lower() == "yes":
                print(get_suggestion(books))
                answer = input("Would you like another suggestion? (yes/no): ")

        case "2":
            new_book = input("Enter book title: ")
            print(add_book(books, new_book))

        case "3":
            book_title = input("Enter book title to remove: ")
            print(remove_book(books, book_title))

        case "4":
            old_title = input("Enter old title: ")
            new_title = input("Enter new title: ")
            print(update_book(books, old_title, new_title))

        case "5":
            print(show_books(books))

        case "6":
            print("Goodbye!")
            operation = 6

        case _:
            print("Invalid option!")

File: dayTwo/Java/BookSuggestionSystem.java
import java.util.ArrayList;
import java.util.Random;
import java.util.Scanner;

public class BookSuggestionSystem {

    public static String getSuggestion(ArrayList<String> books) {

        Random random = new Random();

        int randomBook = random.nextInt(books.size());
        int randomPage = random.nextInt(100) + 1;

        return "\nBook for the Day\n"
                + "Book Title: " + books.get(randomBook)
                + "\nPage: " + randomPage;
    }

    public static String addBook(ArrayList<String> books, String newBook) {

        boolean found = false;

        for (int count = 0; count < books.size(); count++) {

            if (books.get(count).equalsIgnoreCase(newBook)) {
                found = true;
            }
        }

        if (found) {
            return "Book already exists!";
        }

        books.add(newBook);
        return "Book added successfully!";
    }

    public static String removeBook(ArrayList<String> books, String bookTitle) {

        for (int count = 0; count < books.size(); count++) {

            if (books.get(count).equalsIgnoreCase(bookTitle)) {

                books.remove(count);
                return "Book removed successfully!";
            }
        }

        return "Book not found!";
    }

    public static String updateBook(ArrayList<String> books, String oldTitle, String newTitle) {

        for (int count = 0; count < books.size(); count++) {

            if (books.get(count).equalsIgnoreCase(oldTitle)) {

                books.set(count, newTitle);
                return "Book updated successfully!";
            }
        }

        return "Book not found!";
    }

    public static String showBooks(ArrayList<String> books) {

        String allBooks = "\nAll Books\n";

        for (int count = 0; count < books.size(); count++) {

            allBooks = allBooks + (count + 1) + ". "
                    + books.get(count) + "\n";
        }

        return allBooks;
    }

    public static void main(String[] args) {

        Scanner input = new Scanner(System.in);

        ArrayList<String> books = new ArrayList<String>();

        books.add("The Hobbit");
        books.add("The Mystery");
        books.add("Animal Farm");
        books.add("Brave Kingdom");

        int number = 0;

        while (number != 6) {

            System.out.println("\nWelcome to the Book Suggestion System!");
            System.out.println("1. Get Suggestions");
            System.out.println("2. Add Book");
            System.out.println("3. Remove Book");
            System.out.println("4. Update Book");
            System.out.println("5. Show All Books");
            System.out.println("6. Exit");

            System.out.print("Enter action: ");
            number = input.nextInt();
            input.nextLine();


            switch (number) {
                case 1:
                    String answer = "yes";
                    while (answer.equalsIgnoreCase("yes")) {
                        System.out.println(getSuggestion(books));
                        System.out.print("Would you like another suggestion? (yes/no): ");
                        answer = input.nextLine();
                    }
                    break;

                case 2:
                    System.out.print("Enter book title: ");
                    String newBook = input.nextLine();
                    System.out.println(addBook(books, newBook));
                    break;

                case 3:
                    System.out.print("Enter book title to remove: ");
                    String bookTitle = input.nextLine();
                    System.out.println(removeBook(books, bookTitle));
                    break;

                case 4:
                    System.out.print("Enter old title: ");
                    String oldTitle = input.nextLine();
                    System.out.print("Enter new title: ");
                    String newTitle = input.nextLine();
                    System.out.println(updateBook(books, oldTitle, newTitle));
                    break;

                case 5:
                    System.out.println(showBooks(books));
                    break;

                case 6:
                    System.out.println("Goodbye!");
                    break;

                default:
                    System.out.println("Invalid option!");
            }
        }
    }
}

